packages/core/src/testInteg/regionProvider.test.ts

```diff
@@ -0,0 +1,46 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import assert from 'assert'
+import { makeEndpointsProvider } from '../extension'
+import { RegionProvider } from '../shared/regions/regionProvider'
+import globals from '../shared/extensionGlobals'
+import { makeTemporaryToolkitFolder } from '../shared/filesystemUtilities'
+import { fs } from '../shared/fs/fs'
+import path from 'path'
+
+describe('Region Provider', async function () {
+    let tempDir: string
+
+    before(async () => {
+        tempDir = await makeTemporaryToolkitFolder()
```

Comment:
Unless there is a reason, we should be using `TestFolder.create()` from `testUtils.ts` to create a temp folder. And it will also clean itself up, so all you would need to do is use a `beforeEach()`